src/jit/decomposelongs.cpp

```diff
@@ -68,21 +68,39 @@ void DecomposeLongs::DecomposeBlock(BasicBlock* block)
     assert(block == m_compiler->compCurBB); // compCurBB must already be set.
     assert(block->isEmpty() || block->IsLIR());
 
-    m_block = block;
+    m_blockWeight = block->getBBWeight(m_compiler);
+    m_range = &LIR::AsRange(block);
+    DecomposeRange();
+}
+
+void DecomposeLongs::DecomposeRange(Compiler* compiler, unsigned blockWeight, LIR::Range& range)
+{
```

Comment:
> These two methods don't have a block comment describing them.

Good point--I will add header comments.

> Also in general I don't like the names of methods to be overloaded for no good reason.

I'll change the name of the private instance method to `DoDecomposition` or similar.